papers/ring_vrf/overview.tex

```diff
@@ -127,30 +127,31 @@ \section{Protocol overview}
 
 % \smallskip
 
-After these corrections,  our algorithms for a
-{\em ring verifiable random function with associated data} (rVRF-AD)
-now really look like
+Having these concerns in mind, the full name our new notion is \emph{ring verifiable function with additional data}; 
+it has the same api as the \rVRF notion described above with the mention that signing and verifying are now defined as per below. Note that for simplicity and readability,
+we use only the prefix \rVRF:
```

Comment:
I think this parses wrong as the having clause referring to our new notion, but our new notion does not have anything in mind.   I avoid a number of clause types in which I do not really know the correct rules off the top of my head, but of course some diversity is good too, and english is fairly forgiving about mildly incorrect usage.
https://www.sitepoint.com/community/t/start-sentence-with-having/208692/10
https://english.stackexchange.com/questions/375202/starting-a-sentence-with-having-or-when-provided-with

Reply:
You are right, the sentence indeed starts with "Having", however it does not start with a past participle "Having been" which is explicitly mentioned in the first link. Both constructions can be problematic, I agree, but they are not identical.
In order to avoid "having", would the following work better: "Because of these concerns, the full name for our new notion ..."?